addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -13,6 +13,14 @@ const TableOfContent = publicWidget.Widget.extend({
      */
     async start() {
         this._stripNavbarStyles();
+        const $nav = this.$target.find('.s_table_of_content_navbar');
+        let lastVisibleElement;
+        if ($nav.find('a.table_of_content_link[data-visibility-id]').length > 0 ) {
+            lastVisibleElement = $nav.find('a.table_of_content_link:visible:last')
+            if (lastVisibleElement) {
+                lastVisibleElement.css('margin-bottom', '0px');
+            }
+        }
```

Comment:
> You need to find a way to make it so the 0px is applied to the last visible one.

Hello @rdeodoo ,
I currently address the scrollbar issue using this approach. I would like to mention that I am not avoiding the use of jQuery in this process. Initially, it appears to be a visually appealing solution. If this approach proves to be effective and satisfactory, I intend to eventually remove all jQuery-related components.

Reply:
Let's not discuss in multiple review points, we can continue here I guess instead of https://github.com/odoo/odoo/pull/126705#discussion_r1305694932

But the idea was to simulate the visible last as real last element so it got impacted by all the css rules targetting the last one, not only the margin (as QSM said there is a rounder border which I saw but seemed almost invisible but still).
What do you mean by:
> When we switch the language to the visibility language, these elements also become hidden at that point.

When you switch language, you reload the page, which makes the code in 000 kicks in again and should be good?

Reply:
> I guess a better solution would be to remove the invisible elements in the list-group area at start and attach them back in destroy.

Hello @rdeodoo ,
I have implemented the suggested changes, which involve removing the invisible elements from the list-group area at the `start`. There is no need to attach them back in the `destroy`. This is because, when we enter edit mode, the `_generateNav` method is invoked, which generates the navbar for each element. In this case, all the elements become visible, so there is no requirement to reattach them.
However, there is a particular issue related to the scrollbar behavior. Currently, when the last element is invisible, the scrollbar is hidden, but it appears suddenly, almost like a popup, after saving. This behavior is not aligned with our intended user experience. I'm not getting how to handle such case.

Reply:
It's indeed not ideal, but it's more than fine, especially in Odoo 15 and especially since it occurs only when the last element is hidden.
Also, note that it's not only after save but on every page load, you just hit F5 and see the glitch you are speaking of.

Reply:
Note that it would be great to mention that in a comment (the part about not restoring what was deleted).
I will push a final diff so we can ship it